File: backend/src/main/java/com/associazionedisabili/model/Photo.java
package com.associazionedisabili.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "photos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Photo {
 @Id
 @GeneratedValue(strategy = GenerationType.IDENTITY)
 private Long id;

 private String titolo;
 private String descrizione;

 @Column(columnDefinition = "LONGTEXT")
 private String immagine; // Base64 encoded image

 @Column(nullable = false, updatable = false)
 private LocalDateTime createdAt;

 @PrePersist
 protected void onCreate() {
  createdAt = LocalDateTime.now();
 }
}

File: backend/Models/Photo.cs
namespace SoccerDreamFermana.Backend.Models;

public sealed class Photo
{
    public long Id { get; set; }
    public string? Titolo { get; set; }
    public string? Descrizione { get; set; }
    public string? Immagine { get; set; }
    public DateTime CreatedAt { get; set; }
}
